spec/features/admin/members/list_spec.rb

```diff
@@ -3,11 +3,73 @@
 
   before do
     assume_admin_logged_in
-    visit '/admin'
   end
 
   it 'has member name' do
+    visit '/admin'
     click_link('Members', match: :first)
     expect(page).to have_content user.full_name
   end
+
+  describe 'filtering' do
+    context 'when synthetic filter is turned on`' do
+      it 'filters on synthetic users' do
+        synthetic_user = create(:user, :synth)
+        non_synthetic_user = create(:user, first_name: 'Denys')
+
+        visit '/admin/members?filter_member[synthetic]=1'
+
+        expect(page).to have_content synthetic_user.full_name
+        expect(page).to_not have_content non_synthetic_user.full_name
+      end
+    end
+
+    context 'when verified filter is turned on`' do
+      it 'filters on verified users' do
+        verified_user = create(:user, :verified)
+        non_verified_user = create(:user, first_name: 'Denys', verified: false)
+
+        visit '/admin/members?filter_member[verified]=1'
+
+        expect(page).to have_content verified_user.full_name
+        expect(page).to_not have_content non_verified_user.full_name
+      end
+    end
+
+    context 'when verified and synthetic is turned on' do
+      it 'filters on synthetic and verified users' do
+        synthetic_user = create(:user, :synth, :verified)
+        non_synthetic_user = create(:user, verified: false)
+
+        visit '/admin/members?filter_member[synthetic]=1'
+
+        expect(page).to have_content synthetic_user.full_name
+        expect(page).to_not have_content non_synthetic_user.full_name
+      end
+    end
+
+    context 'when speakers filter is turned on`' do
+      it 'filters on speakers users' do
+        speaker = create(:user)
```

Comment:
Can we use `let!` instead of defining variables inside `it` ? It's much more conscious imo